### PR TITLE
Remove on_aws attribute

### PR DESCRIPTION
**Pull request by theseanything**
This PR is to simplify the codebase by removing code used to differentiate between Carrenza and AWS environments. This an artefact from the migration, however that migration is now complete and all applications are deployed to AWS.

This normalises the production and staging environment names with "-aws" suffix. Which can be removed in later PR, however remains for now as deployments from Jenkins are still registered with the suffix.

This removes the "hosted_on_aws" from the deployment API response, but this isn't used by anything.

This PR is scoped to removing the logic differences between deployments for Carrenza or AWS. After this PR, I'll raise another PR to make the environment name consistent i.e. production, staging and integration (dropping the -aws). But can't raise that until, this PR is merged. As part of dropping the -aws suffix, I will also merge the [change to app-deployments](https://github.com/alphagov/govuk-app-deployment/pull/453/files) so new deployments also don't use the -aws suffix.

TLDR; will ultimately end up with production, staging and integration - but just need to this as preliminary work, so I don't break Release and introduce too much downtime.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/controllers/applications_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class ApplicationsController < ApplicationController
 
   include ActionView::Helpers::DateHelper
 
-  ENVIRONMENTS = %w[production staging integration].freeze
+  ENVIRONMENTS = %w[production-aws staging-aws integration].freeze
```
Comment (ChrisBAshton) [inline app/controllers/applications_controller.rb:6]: Quick fly-by comment: can we normalise this as `integration-aws` too? Seems a shame there's still an inconsistency even after this PR.

Comment (ChrisBAshton) [reply app/controllers/applications_controller.rb:6]: Oh wait - looks like that would cause an issue here, wouldn't it? https://github.com/alphagov/govuk-app-deployment/pull/453/files

Comment (theseanything) [reply app/controllers/applications_controller.rb:6]: So we migrated integration completely over to aws earlier than production/staging so we never really used `integration-aws`. However we never dropped the `-aws` suffix for production/staging, so deployments for them have continued to be `production-aws` and `staging-aws`.

This PR is scoped to removing the logic differences between deployments for Carrenza or AWS. After this PR, I'll raise another PR to make the environment name consistent i.e. `production`, `staging` and `integration` (dropping the `-aws`). But can't raise that until, this PR is merged. As part of dropping the `-aws` suffix, I will also merge the [change to app-deployments](https://github.com/alphagov/govuk-app-deployment/pull/453/files) so new deployments also don't use the `-aws` suffix.

TLDR; will ultimately end up with `production`, `staging` and `integration` - but just need to this as preliminary work, so I don't break Release and introduce too much downtime.
